test/specs/components/action.spec.jsx

```diff
@@ -138,258 +144,318 @@ xdescribe('Action', () => {
         });
     });
 
-    describe('buy action with stripe keys and offered state', () => {
-        const props = getBuyProps({
-            uri: 'fallback uri'
-        });
-
-        beforeEach(() => {
-            mockedStore = mockAppStore(sandbox, getStoreState());
-            component = TestUtils.renderIntoDocument(<ActionComponent {...props} />);
-        });
-
-        it('should render a Stripe button', () => {
-            TestUtils.scryRenderedDOMComponentsWithClass(component, 'mockedStripe').length.should.be.eq(1);
-            TestUtils.scryRenderedDOMComponentsWithClass(component, 'mockedLoading').length.should.be.eq(0);
-            TestUtils.scryRenderedDOMComponentsWithClass(component, 'btn-sk-action-paid').length.should.be.eq(0);
-            componentNode.textContent.should.eq('action text');
+    describe('Stripe', () => {
+        const context = getContext({
+            app: {
+                stripe: {
+                    appName: 'app-name',
+                    iconUrl: 'iconUrl'
+                }
+            }
         });
-    });
 
-    describe('buy action with stripe keys and processing state', () => {
-        const props = getBuyProps({
-            state: 'processing'
+        afterEach(() => {
+            appUtils.getIntegration.should.have.been.calledWithMatch(context.app.integrations, 'stripeConnect');
         });
 
-        beforeEach(() => {
-            mockedStore = mockAppStore(sandbox, getStoreState());
-            component = TestUtils.renderIntoDocument(<ActionComponent {...props} />);
-            componentNode = ReactDOM.findDOMNode(component);
-        });
+        describe('buy action with stripe keys and offered state', () => {
+            const props = getBuyProps({
+                uri: 'fallback uri'
+            });
 
-        it('should render a loading element', () => {
-            TestUtils.scryRenderedDOMComponentsWithClass(component, 'mockedStripe').length.should.be.eq(0);
-            TestUtils.scryRenderedDOMComponentsWithClass(component, 'mockedLoading').length.should.be.eq(1);
-            TestUtils.scryRenderedDOMComponentsWithClass(component, 'btn-sk-action-paid').length.should.be.eq(0);
-            componentNode.textContent.should.eq('');
-        });
-    });
+            beforeEach(() => {
+                appUtils.getIntegration.resolves(
```

Comment:
because it returns a truthty value (a promise) and in the component it's used like this 

```
if (this.props.type === 'buy' && stripeIntegration) {
```

then later it passes `stripeIntegration.publicKey` to the StripeCheckout component (which seems to don't care about it to be undefined, or it's actually mocked)